components/more_direct_channels/index.js

```diff
@@ -49,7 +54,9 @@ function mapStateToProps(state, ownProps) {
     } else {
         users = getProfilesInCurrentTeam(state);
     }
-
+    const filteredDirectChannels = getChannelsWithUserProfiles(state).
+        filter((channel) => matchExistsInChannelProfiles(channel.profiles, searchTerm));
+    users = [...users, ...filteredDirectChannels];
```

Comment:
Because of the same things as above, we should use a separate prop here for the group channels since this is returning a new array for users each time `mapStateToProps` runs and because the name is misleading. If we wait to combine the arrays in `MoreDirectChannels`'s render method before passing them to the `Multiselect`, it'll only generate a new array when `MoreDirectChannels` re-renders